test/test_extractor.py

```diff
@@ -26,30 +26,14 @@
     os.path.join(os.path.abspath(os.path.dirname(__file__)), "assets"),
     "cab-test-python3.8.cab",
 )
-BAD_EXE_FILE = os.path.join(
-    os.path.join(os.path.abspath(os.path.dirname(__file__)), "assets"),
-    "empty-file.exe",
-)
-BAD_ZIP_FILE = os.path.join(
-    os.path.join(os.path.abspath(os.path.dirname(__file__)), "assets"),
-    "empty-file.zip",
-)
-BAD_TAR_FILE = os.path.join(
-    os.path.join(os.path.abspath(os.path.dirname(__file__)), "assets"),
-    "empty-file.tar",
-)
-BAD_RPM_FILE = os.path.join(
-    os.path.join(os.path.abspath(os.path.dirname(__file__)), "assets"),
-    "empty-file.rpm",
-)
-BAD_DEB_FILE = os.path.join(
-    os.path.join(os.path.abspath(os.path.dirname(__file__)), "assets"),
-    "empty-file.deb",
-)
-BAD_CAB_FILE = os.path.join(
-    os.path.join(os.path.abspath(os.path.dirname(__file__)), "assets"),
-    "empty-file.cab",
-)
+
+BAD_FILE = dict()
+FILE_EXT = ["tar", "zip", "cab", "exe", "rpm", "deb"]
+for ext in FILE_EXT:
+    BAD_FILE[ext] = os.path.join(
+        os.path.join(os.path.abspath(os.path.dirname(__file__)), "assets"),
+        "empty-file." + ext,
+    )
```

Comment:
```suggestion
    BAD_FILE[ext] = os.path.join(
        os.path.abspath(os.path.dirname(__file__)),
        "assets",
        "empty-file." + ext,
    )
```
   
Just noticed. Do we need couple joins here?

Reply:
Yeah, that's true, just one is enough. It was probably copied from the `CAB_TEST_FILE_PATH` above, it can be changed as well.

I would prefer to do `f"empty-file.{ext}",` instead of `"empty-file." + ext,` but it's just a nit.

And maybe it would make sense to do `from os import path` instead of `import os` to not write `os.path.xxx` every time, just `path.xxx` (you'd need to fix `path` to `dir_path` in loops in `test_extract_file_tar` and `test_extract_file_zip`).

But I'm really moving away from refactoring bad archive file tests to refactoring the whole file 😄

Reply:
Oh, no no not required. I mean, both of them will create empty-file.ext in assets folder. Your suggestion is more readable and simple.

Reply:
@Molkree XD

Reply:
> And maybe it would make sense to do `from os import path` instead of `import os` to not write `os.path.xxx` every time, just `path.xxx` (you'd need to fix `path` to `dir_path` in loops in `test_extract_file_tar` and `test_extract_file_zip`).

there's over 20 occurrences of `os.path.join` alone. Would be great if that could be refactored too but maybe in a different PR

Reply:
sheesh! in the same file?